sql/src/main/java/io/crate/expression/reference/sys/shard/ShardRowContext.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.reference.sys.shard;
+
+import com.google.common.base.Suppliers;
+import io.crate.blob.v2.BlobShard;
+import io.crate.metadata.IndexParts;
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.RelationName;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.index.shard.IndexShard;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.index.store.StoreStats;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+public class ShardRowContext {
```

Comment:
Would this better be named as `ShardContext`? Not sure if there is any row relevant information in there

Reply:
or `SysShardContext` ?

Reply:
this context is used for a shard **row** not for a shard, so I choose this to make it clear ;)

Reply:
I don't see how just `RowContext` will make it more clear, but I'm totally fine with this. not sure what I should document here more that what the class name is telling... (its a context for shard rows)